libyul/optimiser/NameCollector.cpp

```diff
@@ -56,30 +56,62 @@ void ReferencesCounter::operator()(Identifier const& _identifier)
 
 void ReferencesCounter::operator()(FunctionCall const& _funCall)
 {
-	if (m_countWhat == VariablesAndFunctions)
-		++m_references[_funCall.functionName.name];
+	++m_references[_funCall.functionName.name];
 	ASTWalker::operator()(_funCall);
 }
 
-map<YulString, size_t> ReferencesCounter::countReferences(Block const& _block, CountWhat _countWhat)
+map<YulString, size_t> ReferencesCounter::countReferences(Block const& _block)
 {
-	ReferencesCounter counter(_countWhat);
+	ReferencesCounter counter;
 	counter(_block);
-	return counter.references();
```

Comment:
`references()` used to return a `const&` so this used to copy...

Reply:
RVO should kick in here, so I would say no, don't move this explicitly; in fact, doing this explicitly could prevent copy elision from taking place. Also, for future reference, due to some clang tomfoolery, we had to extend the `check_style` script to throw errors when using unqualified moves, so you actually have to use `std::move` instead of just `move`.

Reply:
I think it's still better to use explicit move. The compliers usually complain if it prevents RVO.